CMSgov/mdct-mcr
# revise formTemplate and validationSchema storage

## Description

Needed to restructure how form templates and their corresponding validation schemas are stored. Previously they were stored in separate schema files, but now they are stored inline in ReportJson. In addition, a schema translator utility has been added, which takes schema in JSON format (e.g. `"validation": "text"`) and transforms it to schema in yup format (e.g. `validation: string().typeError().required()`.

Satisfies: https://qmacbis.atlassian.net/browse/MDCT-1957

Changes:
- Moved validation for all forms and fields inside the report JSON
- Created utilities that compile validation from ReportJson and FormField types.
- Created schema translation utility to convert schema from JSON to Yup format.
- Migrated all current forms to using schema translation utility prior to form creation.
- Added functionality for all FormTemplateJson to be stored in the `report` table.

### How to test

- Confirm that flat validation schema json is saved in report table on new report creation.
1. Navigate to dashboard as state user.
2. Create new report.
3. Check for validation schema on form template object in `report` table.
- Confirm field-level validation still works
1. Navigate to any form page as a state user.
2. Attempt to enter invalid data.
3. All field validation schema works.

### Changed Dependencies

n/a

## Code author checklist

## Reviewer checklist (two different people)
